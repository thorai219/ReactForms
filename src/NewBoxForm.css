.form {
  width: 500px;
  margin: 0 auto;
}

.form form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.label {
  margin-right: 20px;
}