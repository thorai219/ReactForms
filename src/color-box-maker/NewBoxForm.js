import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

const NewBoxForm = ({ createBox }) => {
  const initial_state = {
    color: "",
    width: "",
    height: ""
  }
  const [formData, setFormData] = useState(initial_state)
  const handleChange = e => {
    const {name, value} = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }))
  }
  const handleSubmit = e => {
    e.preventDefault();
    createBox({...formData, id: uuidv4()})
    setFormData(initial_state)
  }
  return(
    <div className="form">
      <form onSubmit={handleSubmit}>
        <div>
          <label 
            className="label" 
            htmlFor="color"
          >Color</label>
          <input 
            type="text"
            id="color"
            name="color"
            data-testid="color"
            value={formData.color}
            onChange={handleChange}
          />
        </div>
        <div>
          <label
            className="label" 
            htmlFor="width"
          >Width</label>
          <input 
            type="text"
            id="width"
            data-testid="width"
            name="width"
            value={formData.width}
            onChange={handleChange}
          />
        </div>
        <div>
        <label 
          className="label" 
          htmlFor="height"
        >Height</label>
        <input 
          type="text"
          id="height"
          name="height"
          data-testid="height"
          value={formData.height}
          onChange={handleChange}
        />
        </div>
        <button data-testid="create-btn">Create Box</button>
      </form>
    </div>
  )
}

export default NewBoxForm