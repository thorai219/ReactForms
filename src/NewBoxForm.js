import React, { useState } from 'react';
import uuid from 'uuid/v4';
import './NewBoxForm.css'

const NewBoxForm = () => {
  const initial_state = {
    color: "",
    width: "",
    height: ""
  }
  const [formData, setFormData] = useState(initial_state)
  const handleChange = e => {
    const {name, value} = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }))
  }
  const handleSubmit = e => {
    e.preventDefault();
    createBox({...formData, id: uuid()})
    setFormData(initial_state)
  }
  return(
    <div className="form">
      <form onSubmit={handleSubmit}>
        <div>
          <lable className="label" htmlFor="color">Enter a color</lable>
          <input 
            type="text"
            id="color"
            name="color"
            value={formData.color}
            onChange={handleChange}
          />
        </div>
        <div>
          <lable className="label" htmlFor="width">Enter a width</lable>
          <input 
            type="text"
            id="width"
            name="width"
            value={formData.width}
            onChange={handleChange}
          />
        </div>
        <div>
        <lable className="label" htmlFor="height">Enter a height</lable>
        <input 
          type="text"
          id="height"
          name="height"
          value={formData.height}
          onChange={handleChange}
        />
        </div>
      </form>
    </div>
  )
}

export default NewBoxForm