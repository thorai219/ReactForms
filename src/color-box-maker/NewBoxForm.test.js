import React from "react";
import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it("renders", function() {
  render(<NewBoxForm />);
});

it("match snapshot", function() {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});
