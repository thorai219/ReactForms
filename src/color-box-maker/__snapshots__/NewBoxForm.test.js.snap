// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form"
  >
    <form>
      <div>
        <label
          class="label"
          for="color"
        >
          Color
        </label>
        <input
          data-testid="color"
          id="color"
          name="color"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          class="label"
          for="width"
        >
          Width
        </label>
        <input
          data-testid="width"
          id="width"
          name="width"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          class="label"
          for="height"
        >
          Height
        </label>
        <input
          data-testid="height"
          id="height"
          name="height"
          type="text"
          value=""
        />
      </div>
      <button
        data-testid="create-btn"
      >
        Create Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
